### PR TITLE
Add integration test for ssh-key rotation

### LINKED ISSUE #4396: Add integration test for SSH keypair rotation operation
**How to categorize this issue?**

/kind enhancement
/priority 3

**What would you like to be added**:
Add integration test for SSH keypair rotation operation.

**Why is this needed**:
To prevent any regressions and to make sure that the feature is working on all environments.

### PR DESCRIPTION
**Pull request by ary1992**
**How to categorize this PR?**

/area testing
/kind test

**What this PR does / why we need it**:
Add integration test for ssh-key rotation

**Which issue(s) this PR fixes**:
Fixes #4396 

**Special notes for your reviewer**:

Added test as `beta`. We were not sure whether it can already be marked as `default` as it probably is stable already. WDYT?

**Release note**:

```other operator
NONE
```

### CODE REVIEW

**Comment by gardener-robot-ci-2** [comment]
Thank you @ary1992 for your contribution. Before I can start building your PR, a member of the organization must set the required label(s) {'reviewed/ok-to-test'}. Once started, you can check the build status in the PR checks section below.

**Comment by BeckerMax** [comment]
/reviewed/ok-to-test

**Comment by BeckerMax** [comment]
On a side note: We also considered actually ssh'ing onto a worker node in the test. But that would mean we would need to expose the worker node in the test. This would be different on each infrastructure. In the end this was not added in this test due to the involved complexity.
We could reconsider it once we have an integrationtest setup with provider local.

File: test/integration/shoots/sshrotation/sshkey_rotation.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+)
+
+const (
+	reconcileTimeout = 20 * time.Minute
+)
+
+var _ = ginkgo.Describe("SSH Keypair Rotation", func() {
+	var f = framework.NewShootFramework(nil)
+
+	f.Beta().Serial().CIt("ssh-keypair", func(ctx context.Context) {
```
Comment (ialidzhikov) [inline test/integration/shoots/sshrotation/sshkey_rotation.go]: Shouldn't we move this test rather to `./test/integration/shoots/operations/operations.go` (where the kubeconfig rotation test also lives)?

File: test/integration/shoots/sshrotation/sshkey_rotation.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+		secret := &corev1.Secret{}
+		gomega.Expect(f.SeedClient.Client().Get(ctx, client.ObjectKey{Namespace: f.ShootSeedNamespace(), Name: "ssh-keypair"}, secret)).To(gomega.Succeed())
```
Comment (ialidzhikov) [inline test/integration/shoots/sshrotation/sshkey_rotation.go]:
```suggestion
		gomega.Expect(f.SeedClient.Client().Get(ctx, client.ObjectKey{Namespace: f.ShootSeedNamespace(), Name: v1beta1constants.SecretNameSSHKeyPair}, secret)).To(gomega.Succeed())
```

File: test/integration/shoots/sshrotation/sshkey_rotation.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+		gomega.Expect(f.GetShoot(ctx, f.Shoot)).To(gomega.BeNil())
+		v, ok := f.Shoot.Annotations[v1beta1constants.GardenerOperation]
+		if ok {
+			gomega.Expect(v).NotTo(gomega.Equal(v1beta1constants.ShootOperationRotateSSHKeypair))
+		}
+		gomega.Expect(f.SeedClient.Client().Get(ctx, client.ObjectKey{Namespace: f.ShootSeedNamespace(), Name: "ssh-keypair"}, secret)).To(gomega.Succeed())
+		postRotationPrivateKey := getKeyAndValidate(secret, "id_rsa")
+		postRotationPublicKey := getKeyAndValidate(secret, "id_rsa.pub")
+		gomega.Expect(f.SeedClient.Client().Get(ctx, client.ObjectKey{Namespace: f.ShootSeedNamespace(), Name: "ssh-keypair.old"}, secret)).To(gomega.Succeed())
```
Comment (ialidzhikov) [inline test/integration/shoots/sshrotation/sshkey_rotation.go]:
```suggestion
		gomega.Expect(f.SeedClient.Client().Get(ctx, client.ObjectKey{Namespace: f.ShootSeedNamespace(), Name: v1beta1constants.SecretNameOldSSHKeyPair}, secret)).To(gomega.Succeed())
```

File: test/integration/shoots/sshrotation/sshkey_rotation.go (hunk 1 above)
Comment (ialidzhikov) [inline test/integration/shoots/sshrotation/sshkey_rotation.go]:
```suggestion
	f.Beta().Serial().CIt("should rotate the ssh keypair for a shoot cluster", func(ctx context.Context) {
```

File: test/integration/shoots/sshrotation/sshkey_rotation.go [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+			return nil
+		})
+		gomega.Expect(err).To(gomega.BeNil())
+
+		gomega.Expect(f.GetShoot(ctx, f.Shoot)).To(gomega.BeNil())
+		v, ok := f.Shoot.Annotations[v1beta1constants.GardenerOperation]
+		if ok {
+			gomega.Expect(v).NotTo(gomega.Equal(v1beta1constants.ShootOperationRotateSSHKeypair))
+		}
+		gomega.Expect(f.SeedClient.Client().Get(ctx, client.ObjectKey{Namespace: f.ShootSeedNamespace(), Name: "ssh-keypair"}, secret)).To(gomega.Succeed())
```
Comment (ialidzhikov) [inline test/integration/shoots/sshrotation/sshkey_rotation.go]:
```suggestion
		gomega.Expect(f.SeedClient.Client().Get(ctx, client.ObjectKey{Namespace: f.ShootSeedNamespace(), Name: v1beta1constants.SecretNameSSHKeyPair}, secret)).To(gomega.Succeed())
```

**Review by ialidzhikov — APPROVED** [review]
/lgtm
/squash

**Pull request merged**